cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/CommandUtilTest.java

```diff
@@ -67,8 +70,29 @@ public void testWriteBallerinaToml() throws IOException {
                 readFileAsString(COMMAND_UTIL_RESOURCE_DIR.resolve("expected-ballerina.toml")));
     }
 
+    @Test(description = "Test write new project Dependencies.toml from template dependency-graph.json")
+    public void testWriteDependenciesToml() throws IOException {
```

Comment:
Shall we improve the test case as follows:
1. Publish a template that contains a dependency to package_x. 
2. Add a new major version of package_x which has breaking API changes. 
3. Create a new project with a template and check if the compilation is successful

Reply:
Added